targets/TARGET_NORDIC/TARGET_NRF5x/TARGET_NRF52/serial_api.c

```diff
@@ -331,6 +331,20 @@ static void nordic_nrf5_uart_timeout_handler(uint32_t instance)
         /* No activity detected, no timeout set. */
         nordic_nrf5_uart_state[instance].ticker_is_running = false;
 
+        /* Check if hardware flow control is set and signal sender to stop.
```

Comment:
This block of code is repeated 3 times. Would it make sense to wrap this up in it's own function?

Reply:
It's being called twice. It interrupt handler is convoluted as it is, I don't think adding another function will make it easier to read.